File: programmers/js/basic/181861.js
// 배열의 원소만큼 추가하기
function solution(arr) {
  return arr.reduce((acc, cur) => {
    for (let i = 0; i < cur; i++) {
      acc.push(cur);
    }

    return acc;
  }, []);
}

// 다른 풀이
// function solution(arr) {
//   return arr.reduce((list, num) => [...list, ...new Array(num).fill(num)], []);
// }

File: programmers/java/basic/181861.java
// 배열의 원소만큼 추가하기

import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int num : arr) {
            for (int i = 0; i < num; i++) {
                result.add(num);
            }
        }
        
        return result.stream().mapToInt(v -> v).toArray();
    }
}